### DISCUSSION TITLE
Support for Additional Token Splitting Strategies in Text Preprocessing

### DISCUSSION THREAD

**Discussion by sir3mat** [open-webui/open-webui#5766] [General]
Currently, OpenWebUI RAG's text preprocessing step is limited to using the RecursiveCharacterTextSplitter. However, the resulting text chunks are not always optimal for all use cases, which can lead to suboptimal processing and analysis outcomes.

Would it be feasible to introduce support for configurable token-splitting strategies? This would enable users to select or define more efficient ways to split text based on their specific needs. Additionally, offering the option to integrate a custom API for text splitting during the document upload process through the UI would provide greater flexibility.

What are your thoughts on this proposal?

**Comment by Constey** [comment]
Yes I would also like an option to just enter some seperators in a text field like:
![image](https://github.com/user-attachments/assets/58a95162-cd9c-47fd-b17f-89c5e41c4aa5)

**Comment by haoruilee** [comment]
Any process?

**Comment by martinstafford** [comment]
Agree this would be very useful, allowing structured data like JSON, XML, etc to be split on record boundaries. Currently this kind of data can be split mid record, sometimes resulting in authorative looking information from two different records being included in a single chunk. Would save having to implement a lot of scenarios which might otherwise require an external embedding database and vector search.

Also, being able to configure this per-knowledgebase would be ideal.

**Comment by eanson023** [comment]
I'd like to ask if it's possible to disable the splitting operation, because the data I uploaded is only a small chunk.
